Shahid Nehal/tshirtGrouping.py

```diff
@@ -0,0 +1,184 @@
+'''
+@Author : Shaheed Nehal A
+Application of K-Means Clustering Algorithm to solve the following problem:
+Given 100 random tshirt shirts, create 'k' clusters
+
+Input : A dataset of 100 randomly generated datapoints
+Output : 'k' number of datapoints representing the centroids of the clusters
+
+'''
+
+import random, math, time, numpy, os
+os.system('clear')
+
+'''
+Function to find the distance between 2 points (x, y) and (x_cent, y_cent)
+(x, y) is a point in the dataset
+(x_cent, y_cent) is the centroid
+'''
+def findDistance(x_cent, y_cent, x, y):
+	return int(math.sqrt((x-x_cent)*(x-x_cent) + (y-y_cent)*(y-y_cent)))
+
+
+
+'''
+Function to reset the cluster list after every iteration
+At each iteration , a new cluster is formed. This function initializes
+the clusters to be empty
+'''
+def initializeClusters(k, clusterList, cluster, dist):
+	for i in range(k):
+		clusterList = []
+		cluster.append(clusterList)
+		dist.append(0)
+
+	return clusterList, cluster, dist
+
+
+
+'''
+Function to find the new mean for the newly created cluster
+At each iteration a new cluster is created. This function finds the mean
+of the newly formed cluster of datapoints
+'''
+def findNewMeans(cluster, centroid, k):
+	for i in range(k):
+
+		meanx = 0
+		meany = 0
+
+		if len(cluster[i]) != 0:
+			for point in cluster[i]:
+				meanx += point[0]
+				meany += point[1]
+
+			meanx /= len(cluster[i])
+			meany /= len(cluster[i])
+			centroid[i] = (int(meanx), int(meany))
+
+	return cluster, centroid
+
+
+
+'''
+We perform K-Means Clustering on 100 randomly generated tshirt sizes.
+This function generates 100 random tshirt sizes and creates a dataset called 'data'
+'''
+def generate():
+
+	data = []				#Stores all the t-shirt sizes
+	chest_sizes = []		#Stores the width of the t-shirt
+	shirt_lengths = []		#Stores the length of the t-shirt
+	
+	for tshirt in range(101):
```

Comment:
Please externalize this to a file. It can be a .ini, .json, .properties or any other kinds of file and use a configparser and read these values from there.